<launch>
  <node name="map_server" pkg="map_server" type="map_server" args="$(find collvoid_stage)/world/swarmlab_map.yaml"/>
  <rosparam command="load" file="$(find collvoid_stage)/params/stage_params.yaml"/>
  <rosparam command="load" file="$(find collvoid_stage)/params_created.yaml"/>
  <node pkg="stage_ros" type="stageros" name="stageros" args="$(find collvoid_stage)/world/swarmlab_created.world" respawn="false" output="screen" />


  <include file="$(find collvoid_stage)/launch/amcl_diff_multi.launch">
    <arg name="robot" value="robot_0"/>
  </include>
  <include file="$(find collvoid_stage)/launch/move_base_collvoid.launch">
    <arg name="robot" value="robot_0"/>
    <arg name="type" value="turtle"/>
    <arg name="controlled" value="true"/>
  </include>
  <node pkg="collvoid_controller" type="controller_robots.py" name="controller_robots" ns="robot_0"/>


  <include file="$(find collvoid_stage)/launch/amcl_diff_multi.launch">
    <arg name="robot" value="robot_1"/>
  </include>
  <include file="$(find collvoid_stage)/launch/move_base_collvoid.launch">
    <arg name="robot" value="robot_1"/>
    <arg name="type" value="turtle"/>
    <arg name="controlled" value="true"/>
  </include>
  <node pkg="collvoid_controller" type="controller_robots.py" name="controller_robots" ns="robot_1"/>


  <include file="$(find collvoid_stage)/launch/amcl_diff_multi.launch">
    <arg name="robot" value="robot_2"/>
  </include>
  <include file="$(find collvoid_stage)/launch/move_base_collvoid.launch">
    <arg name="robot" value="robot_2"/>
    <arg name="type" value="turtle"/>
    <arg name="controlled" value="true"/>
  </include>
  <node pkg="collvoid_controller" type="controller_robots.py" name="controller_robots" ns="robot_2"/>


  <node pkg="collvoid_controller" type="controller.py" name="controller" output="screen"/>
  <node pkg="collvoid_controller" type="collvoid_visualizer.py" name="controller_viz" output="screen"/>
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find collvoid_stage)/multi_view.rviz" output="screen" />
</launch>
